pkg/state/fee_validation.go

```diff
@@ -48,8 +48,8 @@ type feeValidationParams struct {
 }
 
 type assetParams struct {
-	quantity   int64
-	decimals   int32
+	quantity   uint64
```

Comment:
I think that `int` types were chosen intentionally for better compatibility with Scala implementation.